pkg/k8s/client.go

```diff
@@ -0,0 +1,76 @@
+package k8s
+
+import (
+	"context"
+	"fmt"
+	"log"
+
+	apiv1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/util/retry"
+)
+
+type Client struct {
+	Clientset kubernetes.Interface
+}
+
+func (c Client) ApplySecret(ctx context.Context, secret *apiv1.Secret) error {
+	_, err := c.createSecret(ctx, secret)
+	if err != nil {
+		log.Print(err)
+		err := retry.RetryOnConflict(retry.DefaultRetry, func() error {
+			// Retrieve the latest version of Secret before attempting update
+			// RetryOnConflict uses exponential backoff to avoid exhausting the apiserver
+			gs, err := c.getSecret(ctx, secret)
+			if err != nil {
+				return err
+			}
+
+			err = c.updateSecret(ctx, gs)
+			if err != nil {
+				return err
+			}
```

Comment:
Any thoughts on doing this instead (this pattern here and a few other places that we don't need the non-error returned value)?

```suggestion
			if err = c.updateSecret(ctx, gs); err != nil {
				return err
			}
```

Reply:
🤷 I like both patterns, I try to lean towards consistency for readability but I also really like limiting scope.  I'm going to apply the suggested changes but I think it'd be a great conversation to continue as we're evolving our internal styles.